### PR TITLE
[GCI-riskycase] Complete Golang-2 task

### PR DESCRIPTION
**Pull request by riskycase**

### CODE REVIEW

File: scripts/inputs.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+INSERT INTO hospital VALUES(1, 'Saint Johns Hospital', 200),
```
Comment (riskycase) [inline scripts/inputs.sql:1]: Added the commands used to insert records

File: main.go [hunk 2]
```diff
@@ -3,11 +3,89 @@ package main
 import (
 	"github.com/gin-gonic/gin"
 	"net/http"
+	"log"
+	//"database/sql"
```
Comment (f4ww4z) [inline main.go]: Just remove this line.

File: main.go [hunk 3]
```diff
@@ -3,11 +3,89 @@ package main
...
+	
+	_ "github.com/lib/pq"
+		"github.com/jmoiron/sqlx"
```
Comment (f4ww4z) [inline main.go]: Can you fix this indent?

File: main.go [hunk 4]
```diff
@@ -3,11 +3,89 @@ package main
...
+ */
+func initDatabase() {
+	url := "postgres://ghuvahbe:8Y7zUWnG1VZSF8vRsZPvo_W-AdD1EVkP@arjuna.db.elephantsql.com:5432/ghuvahbe"
+	var err error //Seperate initialisation allows use of = instead of := in next line
+	//Thus we modify the global variable db instead of declaring a local one with :=
+	db, err = sqlx.Open("postgres", url)
+	if err!= nil {
+		log.Fatalln(err)
+	}
+	db.Ping()
```
Comment (f4ww4z) [inline main.go]: Did you miss error checking this `.Ping()` method?

File: main.go [hunk 5]
```diff
@@ -3,11 +3,89 @@ package main
...
+}
+
+/*
+ * Return a json of all patients
+ */
+func GetAllPatients(c *gin.Context){
+	type Patient struct {
```
Comment (f4ww4z) [inline main.go]: Move this outside the function, so it can be reused.

File: main.go [hunk 6]
```diff
@@ -3,11 +3,89 @@ package main
...
+		Id int `db:"id"`
```
Comment (f4ww4z) [inline main.go]: Also specify the `json` naming for each of [these attributes](https://github.com/riskycase/my-go-app/pull/2/files#diff-7ddfb3e035b42cd70649cc33393fe32cR36-R40).
```suggestion
		Id int `json="id" db:"id"`
```

File: main.go [hunk 7]
```diff
@@ -3,11 +3,89 @@ package main
...
+		Name string `db:"name"`
+		Illness string `db:"illness"`
+		BirthDate string `db:"birth_date"`
+		LocationId int `db:"location_id"`
+	}
+	patients := []Patient{}
+	err := db.Select(&patients, "SELECT * FROM patient")
+	if err!= nil {
+		log.Fatalln(err)
```
Comment (f4ww4z) [inline main.go]: It should panic after logging.

File: main.go [hunk 8]
```diff
@@ -3,11 +3,89 @@ package main
...
+	}
+	c.JSON(http.StatusOK, patients)
+}
+
+/*
+ * Return a json of all hospitals
+ */
+func GetAllHospitals(c *gin.Context){
+	type Hospital struct {
```
Comment (f4ww4z) [inline main.go]: Move this outside the function.

File: main.go [hunk 9]
```diff
@@ -3,11 +3,89 @@ package main
...
+		Id int `db:"id"`
```
Comment (f4ww4z) [inline main.go]: Specify the `json` naming for each attribute.

File: main.go [hunk 10]
```diff
@@ -3,11 +3,89 @@ package main
...
+		log.Fatalln(err)
+	}
+	c.JSON(http.StatusOK, hospitals)
+}
+
+/*
+ * Return a json of all locations
+ */
+func GetAllLocations(c *gin.Context){
+	type Location struct {
```
Comment (f4ww4z) [inline main.go]: Move this struct outside the function.

File: main.go [hunk 11]
```diff
@@ -3,11 +3,89 @@ package main
...
+		Id int `db:"id"`
```
Comment (f4ww4z) [inline main.go]: Specify the `json` naming for each attribute.

File: main.go [hunk 12]
```diff
@@ -3,11 +3,89 @@ package main
...
+		Name string `db:"name"`
+		HospitalId int `db:"hospital_id"`
+	}
+	locations := []Location{}
+	err := db.Select(&locations, "SELECT * FROM location")
+	if err!= nil {
```
Comment (f4ww4z) [inline main.go]:
```suggestion
	if err != nil {
```

File: main.go [hunk 13]
```diff
@@ -3,11 +3,89 @@ package main
...
+	//"database/sql"
+	
+	_ "github.com/lib/pq"
+		"github.com/jmoiron/sqlx"
 )
 
-func main() {
-	router := gin.Default()
-	
+var router = gin.Default()
```
Comment (f4ww4z) [inline main.go]: Initialize this in `initAPI()` method instead. This should be just variable declaration.

File: scripts/inputs.sql (hunk 1 above)
Comment (f4ww4z) [reply scripts/inputs.sql:1]: Well, the `SERIAL` type is auto incrementing, so no need to specify it for new records. Nevermind, take note of this next time. 
```suggestion
INSERT INTO hospital(name, max_patient_amount) VALUES('Saint Johns Hospital', 200),
```

**Review by f4ww4z — COMMENTED** [review]
@riskycase See my comments.

**Review by Akayeshmantha — APPROVED** [review]
LGTM

**Pull request merged**